packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -146,17 +148,19 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
 
     const currentId = id.current;
     setIsCreatingSlices(true);
-    addSlices(newSlices)
-      .then(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Execute modal callback.
-      })
-      .catch(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Show error?
-      });
```

Comment:
i prefer to keep the .then and .catch to make it consistent with the other async ops within the modal, if you want to `setIsCreatingSlices(false)` in a single place you can use `.finally`, but imo it's a good idea to separate both paths because it makes things easier to reason about, even if there's a small duplication in some of the steps

for instance, we forgot to reset the slices and the modal id in the happy path

Reply:
I'm good with that. I changed it back and added the slice reset to the happy path ☑️